test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,287 @@ def forward(self, x):
+            x = self.obs2(x)
+            x = self.fc1(x)
+            x = self.relu(x)
+            return x
+
+    def run_model_and_common_checks(self, model, ex_input, num_epochs, batch_size):
+        # split up data into batches
+        split_up_data = torch.split(ex_input, batch_size)
+        for epoch in range(num_epochs):
+            # reset all model report obs
+            model.apply(
+                lambda module: module.reset_batch_and_epoch_values()
+                if isinstance(module, ModelReportObserver)
+                else None
+            )
+
+            # quick check that a reset occurred
+            self.assertEqual(
+                getattr(model, "obs1").average_batch_activation_range,
+                torch.tensor(float(0)),
+            )
+            self.assertEqual(
+                getattr(model, "obs1").epoch_activation_min, torch.tensor(float("inf"))
+            )
+            self.assertEqual(
+                getattr(model, "obs1").epoch_activation_max, torch.tensor(float("-inf"))
+            )
+
+            # loop through the batches and run through
+            for index, batch in enumerate(split_up_data):
+
+                num_tracked_so_far = getattr(model, "obs1").num_batches_tracked
+                self.assertEqual(num_tracked_so_far, index)
+
+                # get general info about the batch and the model to use later
+                batch_min, batch_max = torch.aminmax(batch)
+                current_average_range = getattr(
+                    model, "obs1"
+                ).average_batch_activation_range
+                current_epoch_min = getattr(model, "obs1").epoch_activation_min
+                current_epoch_max = getattr(model, "obs1").epoch_activation_max
+
+                # print(current_average_range,current_epoch_max,current_epoch_min)
+
+                # run input through
+                model(ex_input)
+
+                # check min and max of epoch is correctly recorded
+                self.assertEqual(
+                    getattr(model, "obs1").epoch_activation_min,
+                    min(current_epoch_min, batch_min),
+                )
+                self.assertEqual(
+                    getattr(model, "obs1").epoch_activation_max,
+                    max(current_epoch_max, batch_max),
+                )
+
+                # check that batches run correctly updated
+                self.assertEqual(
+                    getattr(model, "obs1").num_batches_tracked, num_tracked_so_far + 1
+                )
+
+                # check that average batch activation range updated correctly
+                correct_updated_value = (
+                    current_average_range * num_tracked_so_far + (batch_max - batch_min)
+                ) / (num_tracked_so_far + 1)
+                self.assertEqual(
+                    getattr(model, "obs1").average_batch_activation_range,
+                    correct_updated_value,
+                )
```

Comment:
should we call get_batch_to_epoch_ratio to get the correct ratio and compare with the reference?

Reply:
I just tried doing that and I don't think that this would be the best place to do this because there are special test cases I'm running where the epoch range is 0, so this is just erroring when I try this.

Reply:
Actually I added in and added the check where if diff > 0, then i check the function call

Reply:
sounds good, we should be checking this function I think, looks like that is the main api we provide to user